config/routes.rb

```diff
@@ -1,6 +1,14 @@
 Rails.application.routes.draw do
+  get 'users/:author_id/posts', to: 'posts#index', as: 'posts'
+  get 'users/:author_id/posts/:post_id', to: 'posts#show', as: 'post'
+
+  get 'users', to: 'users#index', as: 'users'
+  get 'users/:id', to: 'users#show', as: 'user'
```

Comment:
- [ ] [OPTIONAL] Kindly write the route this way
```
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```
as it's easier to read and maintain.